source/egg/math/Vector.hh

```diff
@@ -62,7 +62,7 @@ struct Vector2f {
 /// @brief A 3D float vector.
 struct Vector3f {
     constexpr Vector3f(f32 x_, f32 y_, f32 z_) : x(x_), y(y_), z(z_) {}
-    constexpr Vector3f() : x(0.0f), y(0.0f), z(0.0f) {}
+    Vector3f();
```

Comment:
This is a regression - let's define this as constexpr and move it into the header.

Reply:
mmmmm not sure why this change happened... I'll fix

Reply:
OH, because we wanted to leverage the NaN for debug builds. I think we could still do this but in the header